events/ri/src/main/java/org/projectnessie/events/ri/messaging/kafka/json/KafkaJsonEventSubscriber.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.ri.messaging.kafka.json;
+
+import io.quarkus.arc.Unremovable;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.projectnessie.events.api.Event;
+import org.projectnessie.events.ri.messaging.config.MessagingEventSubscribersConfig;
+import org.projectnessie.events.ri.messaging.config.MessagingEventSubscribersConfig.EventSubscriberConfig;
+import org.projectnessie.events.ri.messaging.kafka.AbstractKafkaEventSubscriber;
+import org.projectnessie.events.spi.DelegatingEventSubscriber;
+import org.projectnessie.events.spi.EventSubscriber;
+
+/** An {@link EventSubscriber} that publishes events to a Kafka topic using JSON. */
+@ApplicationScoped
+@Unremovable
+public class KafkaJsonEventSubscriber extends AbstractKafkaEventSubscriber<Event> {
+
+  public static final String CHANNEL = "nessie-kafka-json";
+
+  /**
+   * ServiceLoader shim. This is the class that will be instantiated by the ServiceLoader and
+   * registered with the Events service.
+   *
+   * <p>It delegates to the actual implementation, which is a CDI bean.
+   */
+  public static class ServiceLoaderShim extends DelegatingEventSubscriber {
+
+    public ServiceLoaderShim() {
+      super(CDI.current().select(KafkaJsonEventSubscriber.class).get());
```

Comment:
I like the idea of leveraging Quarkus CDI for subscribers. However, do we really need to go through the `ServiceLoader` in this case? Why not have a `@Default` no-op `EventSubscriber`, which a real one will override using `@Priority`?

Reply:
I thought about that too, but with the caveat that we need to support the case where there are more than one subscriber registered (unlikely, but supported). But it's still doable by injecting `Instance<EventSubscriber>` and iterating over all the discovered beans.

Let's do that in a follow-up PR, does that sound OK to you?

Reply:
+1 for follow-up